stremio-core-protobuf/src/bridge/action.rs

```diff
@@ -209,12 +213,31 @@ impl FromProtobuf<Action> for runtime::Action {
                 Some(action_load::Args::AddonDetails(selected)) => {
                     Action::Load(ActionLoad::AddonDetails(selected.from_protobuf()))
                 }
+                // Board
                 Some(action_load::Args::CatalogsWithExtra(selected)) => {
                     Action::Load(ActionLoad::CatalogsWithExtra(selected.from_protobuf()))
                 }
+                // Board
+                Some(action_load::Args::CatalogsWithExtraAll(_empty)) => {
```

Comment:
What's this action, and why's it's needed since it's just using `ActionLoad::CatalogsWithExtra`? If it's using incorrect action then we should leave this action unimplemented as it might introduce unwanted behaviour where devs would think this is available, but it's working incorrectly actually. Same with `CatalogWithFiltersAll`

Reply:
Yes, so we had some issues with Csharp when trying to pass None for `CatalogWithFilters` as it currently requires you to pass a Selected, no Option there.
I've decided to make the API the same as the 2 actions differ in impl otherwise and remove the confusion of using the actions.

I'm open to suggestions but we need to know beforehand as the CSharp app also needs to be updated.

Reply:
Can't they just create a `Selected` with empty/default values, like done here? Don't think it makes sense to have these actions.

Reply:
I'm ok with either just to make the API the same.

If we change the current setup this would mean a breaking change for the API.


Also, there's a difference between None and empty Selected for CatalogWithFilters

![image](https://github.com/user-attachments/assets/24ba488e-af36-4f6f-a6c1-54b54f943489)

While the action for CatalogWithExtra doesn't even allow a None value, only empty Selected.

![image](https://github.com/user-attachments/assets/ab699525-833f-443e-8c31-d9949b6d8ca5)

Not sure about the differences but I'd like to have a similar API as it was confusing for new devs

Reply:
But these are different actions `CatalogsWithExtra` and `CatalogWithFilters` so they can have different behaviours definitions, whereas this new `CatalogsWithExtraEmpty` is basically a subset of `CatalogsWithExtra`, that's why Id' say it's not needed.
Regarding breaking API, these new actions should not be used anywhere in released apps, so it's no breaking?

Reply:
the breaking change will be `CatalogWithFilters` as it currently does not allow you to pass None.

Reply:
Ok, I agree, let's revert the changes.

For **CatalogWithFilters** - you can pass an empty Selected, which will actually load the first page of the catalog, saving you another dispatch of action to core.
For **CatalogWithExtra** - stays as before, you can pass None to it though.